### PR TITLE
TW-1766: fix 500,404 error in POST request when login

### PR DESCRIPTION
**Pull request by sherlockvn**
## Ticket
**Related issue**
#1766 
## Root cause
**If this is a bug, please provide a brief description of the root cause of the issue**
We create the bootstrap dialog at the wrong time, so the state of bootstrap dialog always be `newSSSSKey`, that means, it will call to POST request to upload new key no matter what even when you have got the recovery key success before in tom server.
## Solution
**Outline the implemented solution, detailing the changes made and how they address the issue**
The solution is direct. We move the creating bootstrap dialog after we have loaded successfully the roomData and accountData from homeserver. This will make sure that the bootstrap state is correct and will not call to upload recoveryKey everytime we log in.
## Impact description
**If this is not a bug, please explain how the changes affect the project**
Can decrypt message with new account now.
## Test recommendations
**Recommendations for how to test this, or anything else you are worried about?**
1. Create a new account (make sure you follow every steps)
2. Create an encrypted chat
3. Send some messages in encrypted chat, then wait for the messages is sent.
4. Log out and log in again
5. You will see that the messages in encrypted chat has been decrypted.
## Pre-merge
**Does anything else need to be done before merging?**
no.
## Resolved
**Attach screenshots or videos demonstrating the changes**
- Web:

https://github.com/linagora/twake-on-matrix/assets/43041967/9bde5adc-7433-431d-b508-c0776258fa99

- Android:
- IOS:

https://github.com/linagora/twake-on-matrix/assets/43041967/76f8ea95-a6fb-44d6-b0e6-b60e3e51a2e5

### CODE REVIEW

**Comment by Te-Z** [comment]
Tested on android, chrome, edge, firefox. LGTM

**Comment by hoangdat** [comment]
For this ticket, please note the cases we use to test
- [ ] Test with brand new twake account, sign in to app, test in encrypted group, then logout and login again
- [ ] Test with matrix account which you have a good backup by self recovery key
- ... 

Please add more cases, we should maintain these test case to verify problem of backup.

**Comment by sherlockvn** [comment]
case 1:

https://github.com/user-attachments/assets/c89ebee7-55b5-4a20-97a3-f0da4ea93fa1

**Comment by sherlockvn** [comment]
case 2: 
https://github.com/user-attachments/assets/f6b42da9-f6fb-4c12-a8ce-258e948aa1ad

**Comment by sherlockvn** [comment]
case 3: import backup key from mobile

https://github.com/user-attachments/assets/bff3396b-569a-4abd-a28d-61efabcb98fb

File: lib/pages/bootstrap/tom_bootstrap_dialog.dart [hunk 1]
```diff
@@ -57,19 +56,17 @@ class TomBootstrapDialogState extends State<TomBootstrapDialog>
...
-  void _createBootstrap() async {
+  void _initData() async {
     WidgetsBinding.instance.addPostFrameCallback((_) async {
       await _loadingData();
     });
   }
 
   Future<void> setupAdditionalDioCacheOption(String userId) async {
-    Logs().d('TomBootstrapDialog::setupAdditionalDioCacheOption: $userId');
+    Logs().i('TomBootstrapDialog::setupAdditionalDioCacheOption: $userId');
```
Comment (Te-Z) [inline lib/pages/bootstrap/tom_bootstrap_dialog.dart:69]: IMO those logs should get back to `Logs().d`

File: lib/pages/bootstrap/bootstrap_dialog.dart [hunk 2]
```diff
@@ -333,25 +331,26 @@ class BootstrapDialogState extends State<BootstrapDialog> {
...
-                              );
-                              if (req.error != null) return;
-                              await KeyVerificationDialog(request: req.result!)
-                                  .show(context);
-                              Navigator.of(context, rootNavigator: false).pop();
-                            },
-                    ),
+                    // TODO: TW-1766: temporary disable right now, because not supported yet
+                    // const SizedBox(height: 16),
+                    // ElevatedButton.icon(
```
Comment (Te-Z) [inline lib/pages/bootstrap/bootstrap_dialog.dart:336]: can you explain why you remove this part ?

**Review by Te-Z — APPROVED** [review]
Tested again on android and chrome. Works fine

File: lib/pages/bootstrap/tom_bootstrap_dialog.dart (hunk 1 above)
Comment (sherlockvn) [reply lib/pages/bootstrap/tom_bootstrap_dialog.dart:69]: don't worry, its only appear when you debug

File: lib/pages/bootstrap/tom_bootstrap_dialog.dart [hunk 3]
```diff
@@ -364,35 +377,35 @@ class TomBootstrapDialogState extends State<TomBootstrapDialog>
...
-                  () =>
-                      _uploadRecoveryKeyState = UploadRecoveryKeyState.initial,
-                );
-              }
-            },
-            (success) => setState(
+    await _deleteRecoveryWordsInteractor.execute().then((either) {
+      _createBootstrap();
+      either.fold(
+        (failure) {
```
Comment (nqhhdev) [inline lib/pages/bootstrap/tom_bootstrap_dialog.dart:383]: Pls test case delete key failed also

Comment (nqhhdev) [reply lib/pages/bootstrap/tom_bootstrap_dialog.dart:383]: This demo

signal-2024-07-17-003020_002

**Review by nqhhdev — APPROVED** [review]
LGTM

**Pull request merged**